File: examples/cpp/overview-simple.cpp
#include <proxsuite/proxqp/dense/dense.hpp>
#include <proxsuite/proxqp/utils/random_qp_problems.hpp>// used for generating a random convex Qp

using namespace proxsuite::proxqp;
using T = double;

int
main()
{
  // generate a QP problem
  T sparsity_factor = 0.15;
  dense::isize dim = 10;
  dense::isize n_eq(dim / 4);
  dense::isize n_in(dim / 4);
  T strong_convexity_factor(1.e-2);
  // we generate a qp, so the function used from helpers.hpp is 
  // in proxqp namespace. The qp is in dense eigen format and 
  // you can control its sparsity ratio and strong convexity factor.
  dense::Model<T> qp = utils::dense_strongly_convex_qp(
                                  dim,
                                  n_eq,
                                  n_in,
                                  sparsity_factor,
                                  strong_convexity_factor);

  // load PROXQP solver with dense backend and solve the problem
  dense::QP<T> Qp(dim, n_eq, n_in);
  Qp.init(qp.H, qp.g, qp.A, qp.b, qp.C, qp.u, qp.l);
  Qp.solve();
}

File: examples/python/overview-simple.py
import proxsuite
from util import generate_mixed_qp


# generate a qp problem
n = 10
H, g, A, b, C, u, l = generate_mixed_qp(n)
n_eq = A.shape[0]
n_in = C.shape[0]

# solve it
qp = proxsuite.proxqp.dense.QP(n, n_eq, n_in)
qp.init(H, g, A, b, C, l, u)
qp.solve()
# print an optimal solution
print("optimal x: {}".format(qp.results.x))
print("optimal y: {}".format(qp.results.y))
print("optimal z: {}".format(qp.results.z))
